### DISCUSSION TITLE
ESP32-S3-DevKitC-1-N8R8 PSRAM notice

### DISCUSSION THREAD

**Discussion by dknubben** [arendst/Tasmota#15935] [Q&A] [answered]
If you own an ESP32-S3-DevKitC-1-N8R8 (or an other variant of it with PSRAM and octal SPI) you need to modify the file boards/esp32s3_8M.json or copy it to ex. esp32s3_8M_opi.json and modify platformio_tasmota_env32.ini (board = NAMEOFBOARD)

It's necessary to set the memory_type.
Otherwise the PSRAM will not be detected.

esp32s3_8M.json:

"build": {
    "arduino":{
      "ldscript": "esp32s3_out.ld",
      **"memory_type": "qspi_opi"**
    },

**Comment by Jason2866** [comment, marked as answer]
We know about this. S3 Flash and PSRAM is a mess. OPI mode does not support fallback modes to slower addressing variants. We managed for ESP32, ESP32-S2 and ESP32-S3 to detect PSRAM during startup and enable it when found. With S3 this is NOT possible since the Flash / PSRAM combination needs to be defined 100% (correctly) BEFORE compiling. A wrong combination ends in a bootlooping device.
To make it worse there is no way to detect (with any existing tool) which combination the S3 device has.

**Reply by rug1024** [reply]
I have this board: https://www.wemos.cc/en/latest/s3/s3.html
All Tasmota-Builds I have tested so far do not recognize the PSRAM (also tried TasmoCompiler Dev).
**Which steps exactly do I have to take to compile a working Tasmota-FW for this board?**
I have very basic knowledge of PlatformIO, so I could accomplish that with some hints (I guess).

**Comment by Jason2866** [comment]
@rug1024 You have to use this [board](https://github.com/arendst/Tasmota/blob/development/boards/esp32s3-qio_opi.json) to compile Tasmota
It is used in this env https://github.com/arendst/Tasmota/blob/development/platformio_tasmota_cenv_sample.ini#L33-L38
So you have to run platformio (after copying `platformio_tasmota_cenv_sample.ini` to `platformio_tasmota_cenv.ini`) with command `platformio run -e tasmota32s3-qio_opi-all`

**Reply by rug1024** [reply]
Thanks, I'll try and report...

**Reply by sfromis** [reply]
There is also a precompiled "unofficial" dev build with this env:
https://github.com/tasmota/install/blob/main/firmware/unofficial/tasmota32s3cdc-qio_opi.bin
Also includes LVGL+HASPmota. It worked for me to get PSRAM detected on another ESP32-S3 board ,

**Reply by rug1024** [reply]
Thanks! Flashed it, 8MB PSRAM detected right away.
Now I just need to compile one with I2C Mic + mp3-streaming support.

**Reply by rug1024** [reply]
Tried several hours with PlatformIO on Windows and also Gitpod Tasmota (platformio run -e tasmota32s3-qio_opi-all), but always getting several compile errors.
Very complicated for me, missing experience here so gotta give up.

**Reply by barbudor** [reply]
Jason has packaged a portable set up of VScode+PIO
Easy to install, only unzip
See https://tasmota.github.io/docs/Visual-Studio-Code/

**Reply by rug1024** [reply]
Thanks @barbudor! Took another day but I got that running now.
Portable version works nicely on my work laptop where I have no admin rights.

**Reply by Jason2866** [reply]
@rug1024 That was my reason for doing the portable build. Work Laptop was waaaay faster than my private Laptop. Since i have a Mac M1 Air this changed dramatically. No Windows Laptop can compile Tasmota as fast as with a Apple M1 (M2).
